### ISSUE TITLE
SRFI 1: List Library

### ISSUE THREAD

**Issue by lassik** [udem-dlteam/libs#22]
https://srfi.schemers.org/srfi-1/srfi-1.html

**Comment by lassik** [comment]
We should clear up the copyright status of the implementations in the repo.

The main file is: https://github.com/udem-dlteam/libs/blob/master/srfi/1/1.scm

It is based on Olin's code from the sample implementation of the SRFI.

@alvatar Can you check whether there is some of your code in the implementation? If so, let's add a copyright line for you to the source file.

**Comment by lassik** [comment]
@feeley Can you do a quick review of `1.scm` for coding style? It's clean and straightforward code but there may be some easy opportunities Gambit-specific optimizations or to make error checking more idiomatic.

**Comment by lassik** [comment]
A lot of code is commented out in that file. We should probably remove it once we have understood what it did. All tests pass without the commented-out code.

**Comment by feeley** [comment]
OK, so now we have the start of the `gambit/list` module which aims to be a superset of SRFI 1.  So we should work mostly in `gambit/list` and later implement SRFI 1 trivially as a library that reexports a subset of the `gambit/list` exports.  The same for `gambit/string` and SRFI 13/152.

**Comment by lassik** [comment]
Perfect! I'm just compiling master.

Are there particular SRFIs you plan to have fully implemented for Gambit's upcoming release?

**Comment by lassik** [comment]
In my experience 1 (list), 13 (string), 132 (sort), 151 (bitwise) are constantly needed. These should be pretty easy to fully implement now that the groundwork is laid.

**Comment by feeley** [comment]
Yes that would be a good start.  The release is still a few weeks in the future.
